tsl/src/continuous_aggs/create.c

```diff
@@ -2911,7 +2943,7 @@ remove_old_and_new_rte_from_query(Query *query)
  * for errors and attempt to rebuild it if required.
  */
 static void
-cagg_rebuild_view_definition(ContinuousAgg *agg, Hypertable *mat_ht)
+cagg_rebuild_view_definition(ContinuousAgg *agg, Hypertable *mat_ht, bool force_rebuild)
```

Comment:
Nit: Since this flag is introduced to check for potential errors specifically in caggs with joins, perhaps it might be better to name it something like `cagg_joins_allowed` or at least add a comment explaining why `force_rebuild` is needed for versions >= 2.10?

Reply:
Thanks @konskov for your comments, but we can not do it only in tsl_cagg_try_repair, because we need to rebuild in cases other than force_rebuild, e.g. v2.7. issue, also in cases where we have force_rebuild we do not need to carry rebuild e.g. cases when there is no join. So we need such changes in cagg_rebuild_view_definition, hence I chose to not make much changes in tsl_cagg_try_repair as well.